File: Java/219. Contains Duplicate II.java
package Java;

import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Set<Integer> myset = new HashSet<Integer>();
        int l = 0, r = k;
        for (int i = l; i <= r && i < nums.length; i++) {
            if (!myset.contains(nums[i])) {
                myset.add(nums[i]);
            } else return true;
        }
        while (true) {
            myset.remove(nums[l]);
            l++;
            r++;
            if (r >= nums.length) break;
            if (!myset.contains(nums[r])) {
                myset.add(nums[r]);
            } else return true;
        }
        return false;
    }
}

File: Python/219. Contains Duplicate II.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        myset = set()
        l = 0
        r = k
        for i in range(min(len(k + 1), len(nums))):
            if nums[i] in myset:
                return True
            else:
                myset.add(nums[i])
        while True:
            myset.remove(nums[l])
            l += 1
            r += 1
            if r > len(nums):
                break
            if nums[r] in myset:
                return True
            else:
                myset.add(nums[r])
        return False
